src/components/landingPage/upload.tsx

```diff
@@ -20,6 +21,10 @@ export const Upload = (props: UploadProps) => {
   const dispatch = useDispatch();
 
   const loadTreeJson = (file: any) => {
+    if (file.size > maxFileSize) {
```

Comment:
~Note that this guardrail will only catch direct file uploads. Going through the `fetch` route would be invisible.~ **[Edit: Continuing reading the PR I see you handled that. My bad!]** (That also has its own set of issues because a chonky file would be slow to load, and waiting ~15sec for it to load before seeing anything happen on screen would be confusing.)